octavia_f5/controller/worker/sync_manager.py

```diff
@@ -187,29 +188,23 @@ def tenant_update(self, network_id, device=None):
         stop=stop_after_attempt(RETRY_ATTEMPTS)
     )
     def tenant_delete(self, network_id, device=None):
-        """ Delete a Tenant
+        """ Delete a Tenant.
+        Instead of HTTP DELETE we POST an empty tenant declaration.
+        It works just as well, but additionally leaves us control over declaration persistency.
 
         :param network_id: network id
         :return: True if success, else False
         """
-        tenant = m_part.get_name(network_id)
+        decl = self._declaration_manager.get_declaration({network_id: []})
 
         if CONF.f5_agent.dry_run:
-            LOG.debug("Faking tenant_delete, tenant='%s', device='%s'", tenant, device)
-            return True
-
-        ret = self.bigip(device).delete(tenants=[tenant])
+            decl.set_action('dry-run')
 
-        """ Instead of running unreliably config sync, we delete the tenant on all devices to ensure
-            L2 cleanup is not blocking next config sync. """
-        if not device and CONF.f5_agent.sync_to_group:
-            try:
-                # Don't fail in case passive device is down, cleanup will handle this case
-                self.passive().delete(tenants=[tenant])
-            except RequestException:
-                pass
+        # No config syncing if we are in migration mode or specificly syncing one device
+        if not CONF.f5_agent.migration and not device and not CONF.f5_agent.sync_to_group:
```

Comment:
Probably meant `and CONF.f5_agent.sync_to_group`?

Reply:
Thanks for noticing this...